core/standalone/src/main/scala/org/apache/openwhisk/standalone/StandaloneDockerSupport.scala

```diff
@@ -105,29 +105,46 @@ object StandaloneDockerSupport {
     prefix + name
   }
 
+  /**
+   * Returns the hostname to access the playground.
+   * It defaults to localhost but it can be overriden
+   * and it is useful when the standalone is run in a container.
+   */
+  def getExternalHostName(): String = {
+    sys.props.get("whisk.standalone.host.external").getOrElse(getLocalHostName())
+  }
+
   /**
    * Returns the address to be used by code running outside of container to connect to
    * server. On non linux setups its 'localhost'. However for Linux setups its the ip used
    * by docker for docker0 network to refer to host system
    */
   def getLocalHostName(): String = {
-    if (SystemUtils.IS_OS_LINUX) hostIpLinux
-    else "localhost"
+    sys.props
+      .get("whisk.standalone.host.name")
+      .getOrElse(if (SystemUtils.IS_OS_LINUX) hostIpLinux
+      else "localhost")
   }
 
   def getLocalHostIp(): String = {
-    if (SystemUtils.IS_OS_MAC || SystemUtils.IS_OS_WINDOWS)
-      hostIpNonLinux
-    else hostIpLinux
+    sys.props
+      .get("whisk.standalone.host.ip")
```

Comment:
This I think is not being used?

Reply:
No. But I added it for completeness. For example a scenario where you may want this in kubernetes.